src/odemis/dataio/hdf5.py

```diff
@@ -611,262 +611,106 @@ def _parse_physical_data(pdgroup, da):
+    """
 
-        try:
-            ds = pdgroup["NumericalAperture"]
-            state = _h5svi_get_state(ds)
-            if state and state[i] in (ST_INVALID, ST_DEFAULT):
-                raise ValueError
-            na = float(ds[i])  # ratio
-            md[model.MD_LENS_NA] = na
-        except (KeyError, IndexError, ValueError):
-            pass
+    """
+    try:
+        ds = pdgroup[name]
 
-        try:
-            ds = pdgroup["AccelerationVoltage"]
-            state = _h5svi_get_state(ds)
-            if state and state[i] in (ST_INVALID, ST_DEFAULT):
-                raise ValueError
-            evolt = float(ds[i])  # V
-            md[model.MD_EBEAM_VOLTAGE] = evolt
-        except (KeyError, IndexError, ValueError):
-            pass
+        state = _h5svi_get_state(ds)
+        if default_status:
+            states = (ST_INVALID, ST_DEFAULT)
+        else:
+            states = ST_INVALID
 
-        try:
-            ds = pdgroup["EmissionCurrent"]
-            state = _h5svi_get_state(ds)
-            if state and state[i] in (ST_INVALID, ST_DEFAULT):
-                raise ValueError
-            ecurrent = float(ds[i])  # A
-            md[model.MD_EBEAM_CURRENT] = ecurrent
-        except (KeyError, IndexError, ValueError):
-            pass
+        if state and state[index] == states:
+            raise ValueError
 
-        try:
-            ds = pdgroup["EmissionCurrentOverTime"]
-            state = _h5svi_get_state(ds)
-            if state and state[i] in (ST_INVALID, ST_DEFAULT):
-                raise ValueError
+        # special case
+        if MD_key is model.MD_EBEAM_CURRENT_TIME:
             # It should contain an array of float of Nx2, where the second dim
             # contains time (s since epoch) & current (A)
-            cot = ds[i].tolist()  # A
-            md[model.MD_EBEAM_CURRENT_TIME] = cot
-        except (KeyError, IndexError, ValueError):
-            pass
-
-        try:
-            ds = pdgroup["PolePosition"]
-            pp = tuple(ds[i])  # px
-            state = _h5svi_get_state(ds)
-            if state and state[i] == ST_INVALID:
-                raise ValueError
-            md[model.MD_AR_POLE] = pp
-        except (KeyError, IndexError, ValueError):
-            pass
-
-        try:
-            ds = pdgroup["XMax"]
-            xm = float(ds[i])  # in m
-            state = _h5svi_get_state(ds)
-            if state and state[i] == ST_INVALID:
-                raise ValueError
-            md[model.MD_AR_XMAX] = xm
-        except (KeyError, IndexError, ValueError):
-            pass
-
-        try:
-            ds = pdgroup["HoleDiameter"]
-            hd = float(ds[i])  # in m
-            state = _h5svi_get_state(ds)
-            if state and state[i] == ST_INVALID:
-                raise ValueError
-            md[model.MD_AR_HOLE_DIAMETER] = hd
-        except (KeyError, IndexError, ValueError):
-            pass
+            md[MD_key] = ds[index].tolist()  # A
```

Comment:
If you use `converter=list`, it's not a special case anymore :-)

Reply:
problem: 

`ds[index].tolist()` converts everything to a list

wheras:

`list(ds[index])` only converts the array to a list, but inside the list are still arrays. 

The test case fails then as it does a assertEqual between array and list items. Can we fix the test case or does it need to be a list in the list? Then we need to handle this exception I think...

Reply:
I see, this metadata is a 2D array...
Then, you need to pass a different converter. In general, you could always pass a lambda function, like `lambda a: a.tolist()`, however, here you can just pass `numpy.ndarray.tolist` .

Reply:
Ah cool, didn't know that this works: 
`numpy.ndarray.tolist(ds[index])`

Reply:
Yes, and that's actually a Python design. A method is just a function which has its first argument (`self`) automatically assigned, but it's also possible to call that function "manually" by using the class. IOW,  the second and third lines are exactly equivalent:
```
obj = Klass()
obj.do()
Klass.do(obj)
```